src/socket.rs

```diff
@@ -336,9 +336,9 @@ impl NlSocket {
             Some(ref mut b) => Nlmsghdr::deserialize(b)?,
             None => unreachable!(),
         };
-        if self.pid.is_none() {
+        if self.pid.is_none() && msg.nl_pid != 0 {
```

Comment:
Please move the socket changes into a separate PR. I'm happy to merge those before this one. I agree that PID tracking is very broken and I'm looking to fix that in v0.5.0 but am happy to turn this off for now in another PR.

Reply:
OK, I've submitted the commit as #57. It's the same one, so it should disappear from the diff here once that one is merged into master.

Reply:
Excellent thank you! I want to think through that one a little more before merging, hence the separate PR. Once we figure that out, I'll ask you to rebase onto master so that you pull in whatever we decide on and then I'll merge this one.